src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -142,51 +198,75 @@ private static ReplicationBatchItem CreateReplicationBatchItem(DocumentsOperatio
                 Name = name,
                 ChangeVector = TableValueToString(context, (int)CountersTable.ChangeVector, ref result.Reader),
                 Value = TableValueToLong((int)CountersTable.Value, ref result.Reader),
+                Collection = TableValueToId(context, (int)CountersTable.Collection, ref result.Reader),
                 TransactionMarker = TableValueToShort((int)CountersTable.TransactionMarker, nameof(ReplicationBatchItem.TransactionMarker), ref result.Reader),
             };
         }
 
-        public void PutCounterFromReplication(DocumentsOperationContext context, string documentId, string name, string changeVector, long value)
+        public enum PutCounterMode
```

Comment:
You don't like the mode switch here or dislike the values of enum that are high level terms? Do you prefer to have separate methods?

Reply:
Don't like the intermixing of responsability. The storage shouldn't make decisions based on the source.
The storage should offer options that match what the source need.

Reply:
The point is passing or not the change vector. In ETL we intentionally don't use it while the replication and import need to pass it. I've exposed the relevant overloads instead of having the notion of the mode:

https://github.com/arekpalinski/ravendb/commit/aa4aea3a39f872b7eb995d5231265b1afec19eb6

Reply:
Changed in: https://github.com/ravendb/ravendb/pull/6713/commits/9d8132eff09b20562c4b88ed74